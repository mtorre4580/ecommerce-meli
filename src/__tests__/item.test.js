import React from 'react';
import Item from '../components/results/item/item';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';

describe('<Item />', () => {

    it('should render snapshot', () => {
        const item = {
            id:'MLA769088925',
            title:'Ipod Touch 5g 32gb',
            price: {
                currency:'ARS',
                amount:3200,
                decimals:null
            },
            picture:'http://mla-s2-p.mlstatic.com/720812-MLA25738240210_072017-I.jpg',
            condition:'used',
            location:'Buenos Aires'
        };
        const component = renderer.create(<Router><Item item={item}/></Router>);
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

});



