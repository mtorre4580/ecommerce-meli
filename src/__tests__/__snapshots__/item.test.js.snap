// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Item /> should render snapshot 1`] = `
<article
  className="item"
>
  <a
    aria-current={false}
    className="item-image"
    href="items/MLA769088925"
    onClick={[Function]}
  >
    <picture
      className="item-image"
    >
      <img
        alt="Ipod Touch 5g 32gb"
        className="d-block item-img"
        src="http://mla-s2-p.mlstatic.com/720812-MLA25738240210_072017-I.jpg"
      />
    </picture>
  </a>
  <div
    className="item-ctn"
  >
    <div
      className="item-header"
    >
      <div
        className="d-flex"
      >
        <p
          className="price"
        >
          <span>
            $
          </span>
          3,200
        </p>
      </div>
      <p
        className="text-location"
      >
        Buenos Aires
      </p>
    </div>
    <a
      aria-current={false}
      className="item-title"
      href="items/MLA769088925"
      onClick={[Function]}
    >
      <h2>
        Ipod Touch 5g 32gb
      </h2>
    </a>
  </div>
</article>
`;
